### DISCUSSION TITLE
Settings files

### DISCUSSION THREAD

**Discussion by Moosems** [pyinstaller/pyinstaller#7016] [Help] [answered]
I want to be able to have a settings file for an application but know that bundles don't like it being internal. I am aware that applications like VSCode and Hyper have settings that can be changed but when looking for the settings files I was unable to find them. How can I create a settings file script that works with all OS's while still being able to be changed?

Works with MacOS and Linux:
```python
import os
from pathlib import Path
user_profile_path = os.environ.get("HOME")
master_location = Path(f"{user_profile_path}/.master.json")
print(master_location)
```

What about Windows? How can I get a master location that works for all OS's?

**Comment by Moosems** [comment]
I realized this is kinda stupid

**Comment by rokm** [comment, marked as answer]
Use something like [`appdirs`](https://pypi.org/project/appdirs/). Other higher-level frameworks also provide their abstractions for user-settings storage, like [QSettings](https://doc.qt.io/qtforpython-5/PySide2/QtCore/QSettings.html) in PySide/PyQt Qt bindings.
